File: ICO/IOITC 2022/APIO 2022/game/cpp/grader.cpp
#include <cstdio>
#include <cstdlib>
#include <vector>
#include "game.h"

namespace {

int read_int() {
  int x;
  if (scanf("%d", &x) != 1) {
    fprintf(stderr, "Error while reading input\n");
    exit(1);
  }
  return x;
}

}  // namespace

int main() {
  int N = read_int();
  int M = read_int();
  int K = read_int();
  std::vector<int> u(M), v(M);
  for (int i = 0; i < M; ++i) {
    u[i] = read_int();
    v[i] = read_int();
  }

  init(N, K);
  int i;
  for (i = 0; i < M; ++i) {
    int answer = add_teleporter(u[i], v[i]);
    if (answer != 0 && answer != 1) {
      i = -1;
      break;
    } else if (answer == 1) {
      break;
    }
  }
  printf("%d\n", i);
  return 0;
}

File: CodeChef Contests/August-Long-2023/grader.py
c = 'codechef'
x = 0
while True:
	try:
		s = input()
		if s == '-1':
			continue

		if len(s) != 8:
			print(s)
			print(c)
			exit()

		for i in range(8):
			if s[i] == c[i]:
				print(s)
				print(c)
				exit()

		x += 1
		if x % 1000 == 0:
			print(x / 1000, '%')
	except EOFError:
		break
